src/NServiceBus.Core/EndpointConfiguration.cs

```diff
@@ -178,6 +178,16 @@ public void OverridePublicReturnAddress(string address)
         }
 
         /// <summary>
+        /// Overrides the local address.
+        /// </summary>
+        /// <param name="transportAddress">The transportAddress.</param>
+        public void OverrideLocalAddress(string transportAddress)
```

Comment:
also, thinking more closely about the name, we should probably keep `OverrideLocalAddress` obsoleted and introduce a new method which doesn't include "Address" in the name, since providing an "address string" (e.g. containing @) would not be valid here.